feat: receive dialog ui

shreyasananth-3 opened the pull request.

shreyasananth-3 committed:
feat: add-new-account-pages-and-assets-modify-existing-components

shreyasananth-3 committed:
feat: add account all pages

shreyasananth-3 committed:
feat: add account all pages

shreyasananth-3 committed:
feat: add account all pages

shreyasananth-3 committed:
feat: add account all pages

shreyasananth-3 committed:
feat: change to be made

advp7 committed:
feat: develop add account sync

shreyasananth-3 committed:
feat: feedback correction updated

advp7 committed:
feat: update add-account

advp7 committed:
Merge branch 'develop' of https://github.com/ShreyasGeetha/cypherock-cysync into feat/dialog-account

advp7 committed:
feat: add connect device dialog

advp7 committed:
Merge branch 'develop' of https://github.com/ShreyasGeetha/cypherock-cysync into feat/dialog-account

shreyasananth-3 committed:
feat: dialog transition

shreyasananth-3 committed:
feat: dialog transition

shreyasananth-3 committed:
feat: dialog transition

shreyasananth-3 committed:
feat: dialog transition

advp7 committed:
feat: add keypress to transition

shreyasananth-3 committed:
feat: dialog transition

shreyasananth-3 committed:
feat: dialog transition

shreyasananth-3 committed:
feat: dialog transition

shreyasananth-3 committed:
Merge pull request #1 from ShreyasGeetha/feat/addAccDiagMgmt

Feat/add acc diag mgmt

shreyasananth-3 committed:
feat: dialog transition

shreyasananth-3 committed:
feat: dialog transition

shreyasananth-3 committed:
feat: dialog transition

shreyasananth-3 committed:
feat: dialog transition

shreyasananth-3 committed:
feat: checkbox issue

shreyasananth-3 committed:
Update LeanBox.tsx

remove multiple onclick and embedded everything under Label tag

shreyasananth-3 committed:
feat: temp commit of the work done on receive

shreyasananth-3 committed:
feat: updating with respect to coding standards

shreyasananth-3 committed:
feat: updating with respect to coding standards

shreyasananth-3 committed:
feat: updating with respect to coding standards

shreyasananth-3 committed:
Update packages/cysync-core/src/dialogs/AddAccountGuide/Dialogs/Device/AddAccountSingleChainDialog.tsx

Co-authored-by: irshadCypherock <80461749+irshadCypherock@users.noreply.github.com>

shreyasananth-3 committed:
feat: receive pages

shreyasananth-3 committed:
Update packages/cysync-core/src/constants/i18n/en.ts

Co-authored-by: irshadCypherock <80461749+irshadCypherock@users.noreply.github.com>

shreyasananth-3 committed:
feat: pr review altered code

shreyasananth-3 committed:
feat: pr review altered code

shreyasananth-3 committed:
feat: merge develop into addAccount

shreyasananth-3 committed:
feat: merge develop into addAccount

shreyasananth-3 committed:
feat: merged addAccount to feat/recieve

shreyasananth-3 committed:
feat: submenu

shreyasananth-3 committed:
feat: removing errors from console

shreyasananth-3 committed:
feat: dropdown hover

shreyasananth-3 committed:
feat: dropdown hover

shreyasananth-3 committed:
feat: dropdown hover

shreyasananth-3 committed:
feat: vertical line component for submenu

shreyasananth-3 committed:
feat: merged updated addAccount to recieve

shreyasananth-3 committed:
feat: submenu dropdown

shreyasananth-3 committed:
feat: final PR changes

shreyasananth-3 committed:
feat: merging latest addAccount changes here

shreyasananth-3 committed:
feat: removing a comment

shreyasananth-3 committed:
Merge branch 'addAccount' into feat/recieve

shreyasananth-3 committed:
feat: final fixes

shreyasananth-3 committed:
Merge branch 'develop' into addAccount

shreyasananth-3 committed:
feat: button disabled fix and dropdown background color on select

shreyasananth-3 committed:
feat: dropdown color

shreyasananth-3 committed:
feat: merged latest addAccount with develop branch

irshadCypherock committed:
Merge branch 'develop' into addAccount

shreyasananth-3 committed:
feat: adding a changeset

shreyasananth-3 committed:
feat: deleting unwanted file syncAccount.tsx

shreyasananth-3 committed:
feat: renamed few vairables

shreyasananth-3 committed:
Merge branch 'develop' into feat/recieve

shreyasananth-3 committed:
feat: merger errors resolved

shreyasananth-3 committed:
feat: removed comments

shreyasananth-3 commented on packages/ui/src/assets/images/common/index.ts:
These icons were added before the generate script was introduced

shreyasananth-3 committed:
feat: removed theme import

irshadCypherock commented on packages/ui/src/themes/theme.styled.ts:
```diff
@@ -69,6 +69,7 @@ export const theme = {
       toggleActive: colors.background.black,
       toggle: colors.background.toggleOff,
       list: colors.background.list,
+      infoBackground: colors.background.infoBackground,
```
Since the object is already named `background`, just `info` is suffisient
```suggestion
      info: colors.background.info,
```

shreyasananth-3 replied:
done

irshadCypherock commented on packages/ui/src/themes/color.styled.ts:
```diff
@@ -61,6 +62,7 @@ export const colors = {
     black: '#000000',
     toggleOff: '#544D43',
     list: '#2C2824',
+    infoBackground: '#4A2D00',
```
Same here

shreyasananth-3 replied:
Changed to Info

irshadCypherock commented on packages/ui/src/components/molecules/Asides/MilestoneAside.tsx:
```diff
@@ -33,7 +33,9 @@ const textColor = (activeTab: number, index: number) => {
   activeTab: number;
-}> = ({ milestones, activeTab }) => (
+  currentDialog: number;
+  totalDialogs: number;
```
Why does it need the currentDialog and totalDialogs?

shreyasananth-3 replied:
Now the logic has changed, have removed it

irshadCypherock commented on packages/ui/src/themes/color.styled.ts:
```diff
@@ -43,6 +43,7 @@ export const colors = {
     errorDark: '#74271C',
     success: '#51C61A',
     white: '#ffffff',
+    red: '#FF624C',
```
This is already there as `error`, also remove any other reference of `red`

shreyasananth-3 replied:
removed

irshadCypherock commented on packages/ui/src/components/atoms/Typography.tsx:
```diff
@@ -29,8 +29,9 @@ export type TypographyColor =
+  | 'info'
   | 'disabled'
-  | 'info';
+  | 'red';
```
error is already present which does the same thing

shreyasananth-3 replied:
removed

irshadCypherock commented on packages/ui/src/assets/images/common/spinner-gold.svg:
Use `Throbber` component for loading

irshadCypherock commented on packages/cysync-core/src/dialogs/Receive/index.tsx:
```diff
@@ -0,0 +1,65 @@
+            <Container width="full" p={2} justify="space-between">
+              <HelpButton text={lang.strings.help} />
+              <CloseButton
+                onClick={() => dispatch(closeDialog('receiveDialog'))}
```
Create an `onClose` function in `useReceiveDialog` and use it everywhere else for closing the dialog box.

shreyasananth-3 replied:
done

irshadCypherock commented on packages/cysync-core/src/dialogs/Receive/context/index.tsx:
```diff
@@ -0,0 +1,147 @@
+    },
+  ];
+
+  const onNext = (tab?: number, dialog?: number) => {
+    if (typeof tab === 'number' && typeof dialog === 'number') {
+      setCurrentTab(tab);
+      setCurrentDialog(dialog);
+    } else if (currentDialog + 1 > tabs[currentTab].dialogs.length - 1) {
+      setCurrentTab(prevProps => Math.min(tabs.length - 1, prevProps + 1));
+      if (currentTab !== tabs.length - 1) {
+        setCurrentDialog(0);
+      }
+    } else {
+      setCurrentDialog(prevProps =>
+        Math.min(tabs[currentTab].dialogs.length - 1, prevProps + 1),
+      );
+    }
+  };
+
+  const onPrevious = () => {
+    if (currentDialog - 1 < 0) {
+      if (currentTab === 0) {
+        setCurrentDialog(0);
+      } else {
+        setCurrentDialog(tabs[currentTab - 1].dialogs.length - 1);
+        setCurrentTab(prevProps => Math.max(0, prevProps - 1));
+      }
+    } else {
+      setCurrentDialog(prevProps => Math.max(0, prevProps - 1));
+    }
+  };
```
Since these functions are states are present in all dialog boxes, it's better if we create a reusable hook for these states and functions instead of redefining everytime.

shreyasananth-3 replied:
created

shreyasananth-3 committed:
feat: few text discrepancies fixed

irshadCypherock commented on packages/ui/src/components/molecules/SnackBar.tsx:
```diff
@@ -0,0 +1,63 @@
+  buttonName: string;
+}
+
+export const SnackBar: React.FC<SnackBarProps> = ({
```
Is this component used anywhere?

shreyasananth-3 replied:
No. Will be taken in future

shreyasananth-3 committed:
feat: data array of InitialiseAccount changed

shreyasananth-3 committed:
feat: changes from UI team feedback

shreyasananth-3 committed:
feat: scrollable dialogbox

shreyasananth-3 committed:
feat: design changes and arrow movements with tab key

shreyasananth-3 committed:
Update dropdown.util.tsx

updated dropdown util so that event propagation is handled except for escape key

shreyasananth-3 committed:
Merge branch 'develop' into addAccount

shreyasananth-3 committed:
feat: changed dropdown and added tab features which are there in addAccount

shreyasananth-3 committed:
feat: removed DialogHeaders

shreyasananth-3 committed:
feat: merging addAccount changes

shreyasananth-3 committed:
feat: refactor

shreyasananth-3 committed:
Merge branch 'develop' into feat/recieve

shreyasananth-3 committed:
feat: pr changes and addAccount changes merged

shreyasananth-3 committed:
feat: optimizing dropdown menu

shreyasananth-3 committed:
feat: change in figma design regarding header

shreyasananth-3 committed:
feat: dropdown escape functionality and focus

shreyasananth-3 committed:
feat: merged latest develop branch

shreyasananth-3 committed:
feat: receive final PR

shreyasananth-3 committed:
feat: receive final PR

irshadCypherock commented on packages/ui/src/components/molecules/LeanBox.tsx:
```diff
@@ -119,17 +153,29 @@ export const LeanBox: FC<LeanBoxProps> = ({
             </Typography>
           )}
-          {rightImage && <ImageContainer>{rightImage}</ImageContainer>}
+          {rightImage && (
+            <ImageContainer>
+              <Image src={rightImage as string} alt="image" />
+            </ImageContainer>
+          )}
```
leftImage and rightImage is a reactNode, It should not be rendered this way.

shreyasananth-3 replied:
done

irshadCypherock commented on packages/ui/src/components/molecules/LeanBox.tsx:
```diff
@@ -104,7 +134,11 @@ export const LeanBox: FC<LeanBoxProps> = ({
           />
         )}
-        {leftImage && <ImageContainer>{leftImage}</ImageContainer>}
+        {leftImage && (
+          <ImageContainer>
+            <Image src={leftImage as string} alt="image" />
+          </ImageContainer>
```
Same here

shreyasananth-3 replied:
done

irshadCypherock commented on packages/ui/src/components/molecules/DropDownListItem.tsx:
```diff
@@ -168,7 +182,7 @@ export const DropDownListItem: FC<DropDownListItemProps> = ({
         {leftImage && (
           <DropDownListItemIconContainer>
-            {leftImage}
+            <Image src={leftImage as string} alt="image" />
```
Use it as ReactNode

shreyasananth-3 replied:
done

irshadCypherock commented on packages/ui/src/components/molecules/Dialog/SuccessDialog.tsx:
```diff
@@ -33,14 +26,6 @@ export const SuccessDialog: React.FC<SuccessDialogProps> = ({
   handleSecButtonClick,
 }) => (
   <DialogBox width={500}>
-    {headerText && (
-      <DialogBoxHeader height={56} width={500}>
-        <Typography variant="fineprint" width="100%" color="muted">
-          <LangDisplay text={headerText} />
-        </Typography>
-      </DialogBoxHeader>
-    )}
-
```
Don't remove this

shreyasananth-3 replied:
reverted

irshadCypherock commented on packages/cysync-core/src/hooks/useTabsAndDialog.ts:
```diff
@@ -0,0 +1,92 @@
+import { ReactNode, useState } from 'react';
```
This file already exists as `useTabsAndDialogs`

shreyasananth-3 replied:
done

irshadCypherock commented on packages/cysync-core/src/hooks/closeDialogBox.ts:
```diff
@@ -0,0 +1,9 @@
+import { DialogName, closeDialog, useAppDispatch } from '~/store';
+
+export const useCloseDialogBox = () => {
+  const dispatch = useAppDispatch();
+  const closeDialogBox = (dialog: DialogName) => {
+    dispatch(closeDialog(dialog));
+  };
+  return closeDialogBox;
+};
```
I don't think we need this. The `onClose` function should be defined in the context itself. Check `AddAccount`

shreyasananth-3 replied:
done

irshadCypherock commented on packages/cysync-core/src/constants/i18n/en.ts:
```diff
@@ -329,6 +330,90 @@ const en = {
+      add: {
+        info: {
+          dialogBox: {
+            header: 'Add new accounts',
+            subheader: 'New Accounts',
+            buttonAddAccount: 'Add Accounts',
+            advanced: 'Show all address types',
+            questionMark: '?',
+          },
+        },
+      },
+      selectCrypto: {
+        info: {
+          dialogBox: {
+            header: 'Select the Wallet & Coins you want to add',
+            subTitle: 'Add a coin/account to wallet',
+            constant: 'Cypherock Red',
+            searchText: 'Search',
+            placeholderText: 'Choose a coin',
+            placeholderWalletText: 'Choose a wallet',
+            text: 'Cypherock Red',
+          },
+        },
+      },
+      addAccountSingleChain: {
+        info: {
+          dialogBox: {
+            header: 'Add new accounts',
+            subheader: 'New Accounts',
+            subheader2: 'Account not yet synced',
+            subheader3: 'Accounts already in portfolio',
+            deselectAllButton: 'Deselect All',
+            selectAllButton: 'Select All',
+            buttonAddAccount: 'Add Accounts',
+            advanced: 'Show all address types',
+            questionMark: '?',
+          },
+        },
+      },
+    },
```
This object is not needed anymore

shreyasananth-3 replied:
done

shreyasananth-3 committed:
feat: pr and testing changes

irshadCypherock commented on packages/ui/src/components/atoms/Input/Input.tsx:
```diff
@@ -79,6 +90,7 @@ export const Input: FC<InputProps & { ref?: ForwardedRef<HTMLInputElement> }> =
           </InputLabel>
         )}
         <InputWrapper>
+          {showImage && <Image src={walletIcon} alt="wallet icon" ml={3} />}
```
Image should be given as props, check how it's done in `DropDownListItem`

irshadCypherock commented on packages/ui/src/components/atoms/InformationBox.tsx:
```diff
@@ -0,0 +1,71 @@
+`;
+
+export const InformationBox: React.FC<InformationBoxProps> = ({
+  imagePath,
+  text,
+  iconImagePath,
```
Do not use image path, use ReactNode instead. Check DropDownListItem

shreyasananth-3 replied:
done

irshadCypherock commented on packages/ui/src/components/molecules/LeanBox.tsx:
```diff
@@ -119,8 +148,14 @@ export const LeanBox: FC<LeanBoxProps> = ({
         )}
         {tag && <Tag>{tag}</Tag>}
         <RightContent>
+          {throbber && <Throbber size={15} strokeWidth={2} />}
```
This should be taken care of by `rightImage`, no need for a separate throbber prop. Check how it's used in `AddAccount`

shreyasananth-3 replied:
done

shreyasananth-3 committed:
feat: final pr

shreyasananth-3 committed:
feat: added change set for final pr

shreyasananth-3 committed:
feat: renaming

shreyasananth-3 committed:
Merge branch 'develop' into feat/recieve
